polars/polars-core/src/frame/groupby/aggregations.rs

```diff
@@ -679,62 +672,99 @@ impl<T: PolarsObject> AggList for ObjectChunked<T> {
     }
 }
 
-pub(crate) trait AggQuantile {
-    fn agg_quantile(
+impl<T> SeriesWrap<ChunkedArray<T>>
+where
+    T: PolarsFloatType + Sync,
+    ChunkedArray<T>: IntoSeries,
+    T::Native: PartialOrd + Num + NumCast + Zero + Simd + std::iter::Sum<T::Native>,
+    <T::Native as Simd>::Simd: std::ops::Add<Output = <T::Native as Simd>::Simd>
+        + arrow::compute::aggregate::Sum<T::Native>
+        + arrow::compute::aggregate::SimdOrd<T::Native>,
+{
+    pub(crate) fn agg_quantile(
         &self,
-        _groups: &[(u32, Vec<u32>)],
-        _quantile: f64,
-        _interpol: QuantileInterpolOptions,
+        groups: &[(u32, Vec<u32>)],
+        quantile: f64,
+        interpol: QuantileInterpolOptions,
     ) -> Option<Series> {
-        None
+        let invalid_quantile = !(0.0..=1.0).contains(&quantile);
+        let ca = &self.0;
+        agg_helper::<T, _>(groups, |(_first, idx)| {
+            debug_assert!(idx.len() <= ca.len());
+            if idx.is_empty() | invalid_quantile {
+                return None;
+            }
+
+            let take = unsafe { self.take_unchecked(idx.iter().map(|i| *i as usize).into()) };
+            take.into_series()
+                .quantile_as_series(quantile, interpol)
+                .unwrap() // Validity of quantile checked
+                .unpack::<T>()
+                .unwrap()
+                .get(0)
+        })
     }
 
-    fn agg_median(&self, _groups: &[(u32, Vec<u32>)]) -> Option<Series> {
-        None
+    pub(crate) fn agg_median(&self, groups: &[(u32, Vec<u32>)]) -> Option<Series> {
+        let ca = &self.0;
+        agg_helper::<T, _>(groups, |(_first, idx)| {
+            debug_assert!(idx.len() <= ca.len());
+            if idx.is_empty() {
+                return None;
+            }
+
+            let take = unsafe { self.take_unchecked(idx.iter().map(|i| *i as usize).into()) };
+            take.into_series()
```

Comment:
see earlier comment

Reply:
these one error out when I remove them, will implement the missing pieces so they can be removed

Reply:
Oh, what was the problem? Maybe I am wrong. I did not compile it.

Reply:
I think there's a missing implementation for ChunkedArray<T: PolarsFloatType>, will have a look at it tonight

Reply:
*_as_series is implemented 3 times for ChunkedArray<T: PolarsIntegerType>, Float32Chunked and Float64Chunked. In this file we have 2 implementations for ChunkedArray<T: PolarsIntegerType> and ChunkedArray<T: PolarsFloatType>. 

So I can either
a) change this file to have the 3 implementations or
b) change polars-core/src/chunked_array/ops/aggregate.rs to have the two implementations.

I tried b) but I'm struggling to convince the compiler that PolarsIntegerType and PolarsFloatType are mutually exclusive so it's throwing errors for a duplicate implementation. 
Can I go with a) ? or do you have a trick for b) ?

Reply:
We can only implement one generic. That is `PolarsNumeric`. For the floats, you must implement for both float64 and float32. It can still dispatch to a function generic over float. That is what I would do.

Reply:
Thanks for your help!